pkg/base/helm.go

```diff
@@ -326,19 +326,30 @@ func helmChartUpstreamPathToBasePaths(upstreamPath string, upstreamFileMap map[s
 			if err := yaml.Unmarshal(content, deps); err != nil {
 				return nil, errors.Wrapf(err, "failed to unmarshal %s", parentChartYaml)
 			}
+			foundSubchartInDeps := false
 			for _, dep := range deps.Dependencies {
-				if dep.Name == subchart && dep.Alias != "" {
-					parts = append(parts, dep.Alias)
+				if dep.Name != subchart {
+					continue
 				}
+				foundSubchartInDeps = true
```

Comment:
i think this needs to be something like this
```golang
if dep.Alias != "" {
	aliases = append(aliases, dep.Alias)
	foundSubchartInDeps = dep.Alias == subchart
} else {
	aliases = append(aliases, dep.Name)
	foundSubchartInDeps = dep.Name == subchart
}
```

Reply:
the subchart is not necessarily found just because the name matches. it either has to match a listed alias, or a name _when_ the alias is not specified.